include/sparrow/buffer/dynamic_bitset/dynamic_bitset.hpp

```diff
@@ -98,18 +100,22 @@ namespace sparrow
               value ? 0u : n
           )
     {
+        base_type::zero_unused_bits();
     }
 
     template <std::integral T>
     constexpr dynamic_bitset<T>::dynamic_bitset(block_type* p, size_type n)
         : base_type(storage_type(p, this->compute_block_count(n)), n)
     {
+        base_type::zero_unused_bits();
     }
 
     template <std::integral T>
     constexpr dynamic_bitset<T>::dynamic_bitset(block_type* p, size_type n, size_type null_count)
         : base_type(storage_type(p, this->compute_block_count(n)), n, null_count)
     {
+        base_type::zero_unused_bits();
+        SPARROW_ASSERT_TRUE(base_type::null_count() == base_type::size() - base_type::count_non_null());
```

Comment:
Should we really have this assert or should we have this in unit test?

Reply:
I think we must keep it to detect when the user creates a bitset in an incorrect state.